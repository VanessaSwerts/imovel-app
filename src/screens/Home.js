import React, { useState, useEffect } from 'react'
import { KeyboardAvoidingView, View, ScrollView, StyleSheet, TextInput, TouchableOpacity, Text, Image } from 'react-native'
import { FontAwesome5 } from '@expo/vector-icons'

import ImovelCard from '../components/ImovelCard'
import colors from '../constants/colors.json'

export default function Home({ navigation }) {

  listTest = {
    names: [
      { 'id': 1, 'address': 'Rua Doutor Teodoro, Centro - São Paulo' },
      { 'id': 2, 'address': 'Rua Portugal, Bela Vista - Belo Horizonte' },
      { 'id': 3, 'address': 'Rua Paula Ney, Vila Mariana - São Paulo' },
      { 'id': 4, 'address': 'Rua Doutor Teodoro, Jardim Europa - São Paulo' },
      { 'id': 5, 'address': 'Rua Doutor Teodoro, Jardim Aeroporto - Alfenas' },
    ]
  }
  return (
    <>
      <KeyboardAvoidingView
        behavior="padding"
        enabled={Platform.OS === 'ios'}
        style={styles.container}
      >
        <View style={styles.headerTitle}>
          <Text numberOfLiner={2} style={styles.title} >Encontre o imóvel ideal para você!</Text>
          <Image style={styles.logo} resizeMode="cover" source={require('../../assets/img/house_agreement.png')} />
        </View>

        <View style={styles.inputContainer}>
          <TextInput
            style={styles.inputText}
            placeholder="Pesquise por localidade..."
            placeholderTextColor="#999"
          />
          <TouchableOpacity style={styles.inputButton}>
            <FontAwesome5
              name="search"
              size={16}
              color={colors["blue"]}
            />
          </TouchableOpacity>
        </View>

        <ScrollView
          scrollsToTop={true}
          showsVerticalScrollIndicator={false}
        >
          {listTest.names.map((item, index) => (
            <View key={item.id} style={styles.cardsContainer}>
              <ImovelCard address={item.address} navigation={navigation} />
            </View>
          ))}
        </ScrollView>

      </KeyboardAvoidingView>

    </>
  )
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors["platinum"],
    paddingVertical: 10,
    paddingHorizontal: 30,
  },

  headerTitle: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginTop: 30,
    marginBottom: 5,
  },

  title: {
    width: 220,
    fontSize: 26,
    fontWeight: "600",
    color: colors['yellow'],
    textAlign: 'left'
  },

  logo: {
    height: 60,
    width: 60,
  },

  inputContainer: {
    backgroundColor: '#FFF',
    alignSelf: 'stretch',
    height: 46,
    borderWidth: 1,
    borderColor: colors["blue"],
    borderRadius: 4,
    marginTop: 10,
    marginBottom: 15,
    paddingHorizontal: 12,
    flexDirection: "row",
    justifyContent: 'space-between',
    shadowOpacity: 0.2,
    shadowRadius: 2,
    shadowOffset: {
      height: 0,
      width: 0,
    },
    elevation: 2,
  },

  inputText: {
    paddingHorizontal: 10,
    paddingVertical: 10,
  },

  inputButton: {
    alignSelf: 'center',
  },

  cardsContainer: {
    flex: 1,
    alignSelf: 'stretch',
    justifyContent: 'center',
    marginBottom: 15,
  },

})