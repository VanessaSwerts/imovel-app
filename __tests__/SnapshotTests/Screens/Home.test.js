import React from 'react'
import { act, create } from 'react-test-renderer'
import Home from '../../../src/screens/Home'

jest.useFakeTimers()
jest.mock('@expo/vector-icons/FontAwesome5', () => 'Icon')
jest.mock('@react-native-community/async-storage', () => 'AsyncStorage')
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: jest.fn(),
    })
  }
})

describe('Home test', () => {

  it('render Home screen correctly', () => {
    act(() => {
      tree = create(
        <Home />
      )
    })

    expect(tree.toJSON()).toMatchSnapshot();
  })
})