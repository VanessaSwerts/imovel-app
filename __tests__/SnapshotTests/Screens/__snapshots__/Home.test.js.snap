// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test render Home screen correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#E5E5E5",
        "flex": 1,
        "paddingHorizontal": 30,
        "paddingVertical": 10,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 5,
        "marginTop": 30,
      }
    }
  >
    <Text
      numberOfLiner={2}
      style={
        Object {
          "color": "#FCA311",
          "fontSize": 26,
          "fontWeight": "600",
          "textAlign": "left",
          "width": 220,
        }
      }
    >
      Encontre o imóvel ideal para você!
    </Text>
    <Image
      resizeMode="cover"
      source={1}
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFF",
        "borderColor": "#14213D",
        "borderRadius": 4,
        "borderWidth": 1,
        "elevation": 2,
        "flexDirection": "row",
        "height": 46,
        "justifyContent": "space-between",
        "marginTop": 10,
        "paddingHorizontal": 12,
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0.2,
        "shadowRadius": 2,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      placeholder="Pesquise por localidade..."
      placeholderTextColor="#999"
      rejectResponderTermination={true}
      style={
        Object {
          "paddingHorizontal": 10,
          "paddingVertical": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "opacity": 1,
        }
      }
    >
      <Icon
        color="#14213D"
        name="search"
        size={16}
      />
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onRefresh={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
        />
      }
      refreshing={false}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
